rasa/core/domain.py

```diff
@@ -344,6 +367,11 @@ def collect_templates(
         templates = {}
         for template_key, template_variations in yml_templates.items():
             validated_variations = []
+            if template_variations is None:
+                raise InvalidDomain(
+                    "Utterance {} has no '- text: ' attribute".format(template_key)
```

Comment:
Now it should require `- text` OR `- custom`. I think the code should still work for this, but the warning message should change
```suggestion
                    "Utterance '{}' has no '- text: ' attribute".format(template_key)
```